### PR TITLE
doc: remove duplicate definitions of Doxygen groups

### PR DESCRIPTION
**Pull request by aabadie**
### Contribution description

This PR is fixing all duplicate definitions of Doxygen groups reported by the doccheck script PRed in #11813.

The last commit of this PR is updating the script so that it returns an error when duplicate Doxygen group definitions are reported by the script.

### Testing procedure

- A green CI: `make static-test` returns with no error
- Doxygen documentation is correct: run `make doc` and verify

### Issues/PRs references

~Based on #11813 but~ can also be merged directly.

### CODE REVIEW

File: pkg/lwip/include/lwip.h [hunk 1]
```diff
@@ -7,13 +7,7 @@
  */
 
 /**
- * @defgroup    pkg_lwip    lwIP
- * @ingroup     pkg
```
Comment (miri64) [inline pkg/lwip/include/lwip.h]: This is removed without replacement, causing lwIP not to show up under `pkg` anymore.

Comment (aabadie) [reply pkg/lwip/include/lwip.h]: This is already the case in master (and it's related to #8664).

**Comment by aabadie** [comment]
> I find a minor difference.

Can you point to that difference ? For the moment, I can't any reference to lwip on doc.riot-os.org.

**Comment by miri64** [comment]
> > I find a minor difference.
> 
> Can you point to that difference ? For the moment, I can't any reference to lwip on doc.riot-os.org.

Mh... maybe I've mis-seen something, because now I can confirm that

File: pkg/lwip/include/lwip.h (hunk 1 above)
Comment (miri64) [reply pkg/lwip/include/lwip.h]: You are right

**Comment by miri64** [comment]
Can you remove 7b0ac83ac915ae9aee913198bfc29b4bf2e1cd9a so we can do as suggested in op?

>  but can also be merged directly.

**Comment by miri64** [comment]
(I did not review the bash code  yet and it is still under review in #11813.

**Comment by miri64** [comment]
I meant 7b0ac83... (can't edit the post)

**Comment by aabadie** [comment]
Thanks for having a look @miri64. Is it ok if I cut out 7b0ac83 and 0b46644 and apply the latter to #11813 directly ?

**Comment by miri64** [comment]
Yepp of course!

**Comment by aabadie** [comment]
Done :)

**Comment by miri64** [comment]
Since you also rebased to master, I will need to look into this again in detail. I'll do so tomorrow.

**Comment by miri64** [comment]
Range diff is okay:

```
$ git range-diff upstream/master 0b46644 HEAD
 1:  7b0ac83ac9 <  -:  ---------- tools/doccheck: add multiple group definition check
 2:  fb738cc272 =  1:  248f8d3284 boards/esp8266: deduplicate board doxygen group definition
 3:  814445c3cd =  2:  01814dec19 boards/nucleo-*: deduplicate doxygen board group definitions
 4:  dfa07f2dfd =  3:  6bf9290ed5 boards/stm32f769i-disco: deduplicate doxygen group definition
 5:  746174be71 =  4:  570f308551 cpu/efm32: deduplicate cpu_efm32 group definition
 6:  05a0b8ef4e =  5:  9e9ef3293d sys/sha1: deduplicate sys_hashes_sha1 group definition
 7:  34dbaa92ff =  6:  998fd6bcc9 sys/pthread: deduplicate pthread group definition
 8:  6be82342fd =  7:  bd816f3fb1 pkg/lwip: cleanup doxygen documentation
 9:  1411196e60 =  8:  90e136405e boards/common/esp: deduplicate common esp boards group definitions
10:  74b68e61ff =  9:  8196ea7e58 boards/arduino-leonardo: remove duplicate doc group definition
11:  ce0a9320e6 = 10:  fdfd3133f7 core/sched_native: fix duplicate core_sched group definition
12:  db32d467cb = 11:  f922df8c99 net/gnrc/sock: remove duplicate net_gnrc_sock group definition
13:  0c99f9f5b8 = 12:  e352b54a9b net/gnrc/tcp: remove duplicate net_gnrc_tcp group definitions
14:  0b46644f23 <  -:  ---------- tools/doccheck: enable error for multiple group defines
```

**Comment by aabadie** [comment]
Thanks @miri64 !

**Pull request merged**